doc/spring-boot-plugin-dev.md

```diff
@@ -0,0 +1,318 @@
+and inject their configuration into the Spring application context at runtime.
+In order to do so, the core application must provide a hook for adding a Spring [context configuration file](https://docs.spring.io/spring-framework/docs/4.2.x/spring-framework-reference/html/xsd-configuration.html)
+which loads the plugin configuration and beans from any number of separate plugin JARs.
+The [external context configuration](#external-context-configuration) section of this document explains this in detail.
+
+For plugins whose features are "visible" to the core application (for example, an integration with an external database),
+we define the required behaviour as a Kotlin interface in the core application, and implement that interface in the plugin library.
+In this way, classes in the core application, such as controllers or services, can use the features of the interface at run time
+without any knowledge of the plugin's implementation at compile time.
+
+However, the approach described in this document also works for adding entirely new behaviour to the application,
+such as new request handlers, message converters, MVC configurers, and so on.
+Effectively, anything that can be added to the Spring application context can be written as a plugin.
+
+## A Simplified Example
+
+In order to illustrate the plugin approach, we will refer to an example project: [Spring Plugin Example](https://github.com/simonoakesepimorphics/spring-plugin-example),
+which you can download, run and modify yourself.
+This example has the following characteristics:
+
+* The core "greeting" application has one endpoint, `/greeting`, accepting a required `name` parameter, and an optional `language` parameter.
+* The core implementation returns a default greeting addressed to the specified name.
+* A "language plugin" interface defines a method for selecting the greeting format for the specified language.
+* If a language plugin implementation is present in the application context and the language is specified, the greeting controller uses a greeting format obtained from the language plugin instead of the default format.
+* The language plugin contains an implementation which reads the greeting format for each language from an embedded JSON resource.
+* The language plugin introduces a dependency on the [Jackson](https://github.com/FasterXML/jackson) library for JSON deserialization.
+
+We define the plugin behaviour as a Kotlin interface in the core application ([LanguagePlugin](https://github.com/simonoakesepimorphics/spring-plugin-example/blob/main/greeting-core/src/main/kotlin/com/epimorphics/greeting/LanguagePlugin.kt)),
+and the plugin library provides an implementation of that interface ([JsonLanguagePlugin](https://github.com/simonoakesepimorphics/spring-plugin-example/blob/main/greeting-language-plugin/src/main/kotlin/com/epimorphics/greeting/JsonLanguagePlugin.kt)).
```

Comment:
These classes are small enough and I presume you will have them inline in the actual blog post so that reader doesn't have to navigate to a new page.